src/components/ha-assist-chat.ts

```diff
@@ -328,6 +330,43 @@ export class HaAssistChat extends LitElement {
             this._addMessage(hassMessage);
           }
 
+          if (event.type === "intent-progress") {
+            const delta = event.data.chat_log_delta;
+
+            // new message
+            if (delta.role) {
+              // If currentDeltaRole exists, it means we're receiving our
+              // second or later message. Let's add it to the chat.
+              if (currentDeltaRole && delta.role && hassMessage.text !== "…") {
```

Comment:
OK so can `hassMessage.text` ever be `...` in this case? If so, then it seems like we drop the delta. (If not then just always overwrite instead)

Reply:
I'm still now confused by:
- we update `hassmessage.text`
- then replace `hassmessage` entirely

Reply:
yes it can be if the first message only contained tool calls and no content.

Reply:
if we don't have this check, we will end up with an empty speech bubble (because the Assistant message had no content.

Reply:
- User to Assistant: Turn on the lights in the living room
- Assistant to HA: tool-call: turn on lights
- HA to Assistant: tool-call-result: lights turned on
- Assistant to User: the lights have been turned on

Reply:
What is this doing? Why update .text to then overwrite the entire object?
```
hassMessage.text = ...
hassMessage = 
```

Reply:
hassMessage is a pointer to an object that is already being rendered. When we update text and call request update, it re-renders it. Add message also rerenders it. 

While we’re receiving content, we always have … at the end of the message that we’re processing (see video). When a second message comes in, we remove the … from the current last message and then add a new message.

Reply:
Thanks, got it.